src/main/java/uk/ac/ebi/subs/validator/coordinator/AssayDataValidationMessageEnvelopeExpander.java

```diff
@@ -31,27 +31,29 @@ public AssayDataValidationMessageEnvelopeExpander(AssayRepository assayRepositor
     @Override
     public void expandEnvelope(AssayDataValidationMessageEnvelope assayDataValidationMessageEnvelope) {
         final List<AssayRef> assayRefs = assayDataValidationMessageEnvelope.getEntityToValidate().getAssayRefs();
-        final List<Submittable<uk.ac.ebi.subs.data.submittable.Assay>> assays = new ArrayList<>();
-        final List<Submittable<Protocol>> protocols = new ArrayList<>();
+        if (assayRefs != null && !assayRefs.isEmpty()) {
```

Comment:
there is no else block against this condition. I hope we have handled null assayRefs case some where else.

Reply:
If that condition is false, then we won't fill in the assays and protocols in that envelope object.  They would be just empty `Collection`s.